client/js/src/utils/submit.ts

```diff
@@ -497,7 +497,10 @@ export function submit(
 						? `https://moon-${hostname.split(".")[1]}.${hfhubdev}`
 						: `https://huggingface.co`;
 					const zerogpu_auth_promise =
-						dependency.zerogpu && window.parent != window && config.space_id
+						dependency.zerogpu &&
+						typeof window !== "undefined" &&
+						window.parent != window &&
+						config.space_id
 							? post_message<Headers>("zerogpu-headers", origin)
 							: Promise.resolve(null);
```

Comment:
This is getting pretty verbose now. Could we break this out into an `if` block instead?